src/package-scanner.ts

```diff
@@ -114,8 +117,17 @@ export class PackageScanner {
    * Scan each document in a package manually.
    */
   scanPackageManually() {
+    // Scan top-level entrypoints first, to make sure thier dependencies are
+    // properly converted to JS modules as well.
     for (const document of this.getPackageHtmlDocuments()) {
-      this.scanDocument(document);
+      if (this.topLevelEntrypoints.has(
+              this.urlHandler.getDocumentUrl(document))) {
+        this.scanDocument(document, 'js-module');
+      }
+    }
+    // Scan all other documents, to be converted as top-level HTML files.
+    for (const document of this.getPackageHtmlDocuments()) {
```

Comment:
Right now this iterates over all documents and tries to scan each one to an html-document, which sounds wrong. This PR does a good job of moving off the side-effects of scanDocument, could we go even further and add a `shouldScanDocument()` check here as well?

Reply:
added